parsec/backend/events.py

```diff
@@ -73,7 +74,8 @@ def _on_roles_updated(
                     EventsListenRepOkRealmRolesUpdated(realm_id, role)
                 )
             except trio.WouldBlock:
-                client_ctx.logger.warning("dropping event (queue is full)")
+                client_ctx.event_bus_ctx.clear()
+                abort(409)
```

Comment:
This won't work: `_on_role_updated` is a callback being called through the event bus by the coroutine responsible to listening the events coming from postgresql. Hence raising an abort exception doesn't work given we are not in the context of an event_listen request !

Instead we should close the cancel_scope for the client (this will close the client connection in a clean way)
https://github.com/Scille/parsec-cloud/blob/3adc3c3a4978ff353582b851e094a9a35d7f6684/parsec/backend/asgi/ws.py#L68

So what is done for the `_on_revoked` event for instance. Of course the `_on_revoked` event has directly access to this cancel_scope, while the callbacks in events.py don't.
So we need to add this cancel_scope to the client_ctx object:

```python

    if isinstance(client_ctx, AuthenticatedClientContext):
        with trio.CancelScope() as client_ctx.cancel_scope:
            with backend.event_bus.connection_context() as client_ctx.event_bus_ctx:
                [...]
```
```

Reply:
And we don't need to clear the event queue: it will be automatically done by garbage collector when the client connection is terminated (and triggering the cancel scope is idempotent)